src/shared/serialize.c

```diff
@@ -480,12 +480,32 @@ int deserialize_pidref(FDSet *fds, const char *value, PidRef *ret) {
 
         e = startswith(value, "@");
         if (e) {
-                int fd = deserialize_fd(fds, e);
+                _cleanup_free_ char *fdstr = NULL, *pidstr = NULL;
+                _cleanup_close_ int fd = -EBADF;
+
+                r = extract_many_words(&e, ":", /* flags = */ 0, &fdstr, &pidstr);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to deserialize pidref '%s': %m", e);
+                if (r == 0)
+                        return log_debug_errno(SYNTHETIC_ERRNO(EINVAL), "Cannot deserialize pidref from empty string.");
```

Comment:
can you please also catch the > 2 case and error out?

Reply:
Hmm, I don't see the need. `extract_many_works` counts the expected number of strings, so it can't return > 2 even if there's more to parse.

Reply:
indeed, ignore me.

assert() might still be good to validate this.